content/reference/stacks.md

```diff
@@ -0,0 +1,277 @@
+* For any **global** resources, such as Account Settings, IAM roles and policies, DNS records, or similar, the `environment` for the stack should be `gbl`.
+  * For example, to deploy the `iam-delegated-roles` component (where all resources are global and not associated with an AWS region) to your production account, you should utilize a `stacks/gbl-prod.yaml` stack file.
+
+## Shared Stack Configuration
+
+When you have configuration that you want to share across various stacks, use catalogs. Catalogs are the SweetOps term for shared, reusable configuration.
+
+By convention, all shared configuration for stacks is put in the `stacks/catalogs/` folder, which can then be used in the root `stacks/` stack files via `import`.
+
+There are a few types of shared configuration which are worth noting:
+
+* **Global Catalogs**: For any configuration to share across **all** stacks.
+  * For example, you create a `stacks/catalogs/globals.yaml` file and utilize `import` wherever you need that catalog.
+* **Environment Catalogs**: For any configuration you want to share across `environment` boundaries.
+  * For example, to share configuration across `ue2-stage.yaml` and `uw2-stage.yaml` stacks, you create a `stacks/catalogs/stage-globals.yaml` file and utilize `import` in both the `ue2-stage.yaml` and `uw2-stage.yaml` stacks to pull in that catalog.
+* **Stage Catalogs**: For any configuration that you want to share across `stage` boundaries.
+  * For example, to share configuration across `ue2-dev.yaml`, `ue2-stage.yaml`, and `ue2-prod.yaml` stacks, you create a `stacks/catalogs/ue2-globals.yaml` file and `import` that catalog in the respective `dev`, `stage`, and `prod` stacks.
+* **Base Components**: For any configuration that you want to share across all instances of a component.
+  * For example, you're using the `eks` component and you want to ensure all of you EKS clusters are using the same Kubernetes version, you create a `stacks/catalogs/eks-globals.yaml` file which specifies the `eks` component's `vars.kubernetes_version`. You can then `import` that base component configuration in any stack file that uses the `eks` component.
+  * More information in the below section.
+
+## Component Inheritance
+
+Using component catalogs, you can define default values for all instances of a component across your stacks. But it is also important to note that you can provide default values for multiple instances of a component in a single stack using the component inheritance pattern via the `component` key / value:
+
+```yaml
+# stacks/catalogs/s3-globals.yaml
+components:
+  terraform:
+    s3-bucket:
+      vars:
+        enabled: false
+        user_enabled: false
+        acl: private
+        grants: null
+        versioning_enabled: true
+```
+
+```yaml
+# stacks/uw2-dev.yaml
+imports:
+  - catalogs/s3-globals
+  - catalogs/dev-globals
+  - catalogs/uw2-globals
```

Comment:
```suggestion
  - catalog/component/s3-bucket
  - catalog/dev/globals
  - catalog/uw2/globals
```

Reply:
@osterman Aha interesting. I was already doing a bit of pattern design from some not-so-defined / scattered patterns I was seeing in some of the client repos and it seems you're doubling down on that here. I like it! Once this PR merges, let's be sure to let the full team know about this document and hold people to these patterns!

Reply:
Yes, I'm making an executive decision here that I want to use the `catalog/` pattern for imports with stacks.